code/WorkInProgress/MechanicMadness.dm

```diff
@@ -101,46 +141,61 @@
 				return 1
 		else if (src.open || !istype(W,/obj/item/mechanics))
 			..()
+			src.updateIcon()
 		return 1
-	proc/updateIcon()
-		if(src.welded)
-			src.icon_state=initial(src.icon_state)+"_w"
+	proc
+		updateIcon()
+			if(src.welded)
+				src.icon_state=initial(src.icon_state)+"_w"
+			else if(src.open)
+				// ugly warning, the istype() is 1 when there's a trigger in the container\
+					it subtracts 1 from the list of contents when there's a trigger \
+					doing arithmatic on bools is probably not good!
```

Comment:
oh god what is this comment please save us and make it a multiline comment please i beg

Reply:
backslash continues comments in DM loopy land

Reply:
i know but it breaks syntax highlighting